
<launch>
    <include file="$(find mavros)/launch/apm.launch"/>

    <node pkg="kkctbn2019" name="motor_controller" type="motor_controller" output="screen"/>

    <node pkg="kkctbn2019" name="mode" type="mode.py" output="screen"/>

    <node pkg="kkctbn2019" name="receiver" type="receiver.py" output="screen"/>
    <node pkg="kkctbn2019" name="auto_control" type="auto_control.py" output="screen"/>
    <node pkg="kkctbn2019" name="image_proccessing" type="image_proccessing.py" output="screen"/>

    <node pkg="kkctbn2019" name="setpoint" type="setpoint" output="screen"/>

    <node pkg="kkctbn2019" name="pwm_override" type="pwm_override.py" output="screen"/>

    <node pkg="kkctbn2019" name="server" type="server.py" output="screen"/>

    <node pkg="kkctbn2019" name="throttle_pwm" type="throttle_pwm.py" output="screen"/>

    <node respawn="true" pkg="joy" type="joy_node" name="turtle_joy" >
      <param name="dev" type="string" value="/dev/input/js0" />
      <param name="deadzone" value="0.12" />
    </node>
    
    <node name="controller" pkg="pid" type="controller" output="screen">
      <param name="Kp" value="1.4" />
      <param name="Ki" value="0.1" />
      <param name="Kd" value="0" />
      <param name="upper_limit" value="1000.0" />
      <param name="lower_limit" value="-1000.0" />
      <param name="windup_limit" value="100.0" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
    </node>
</launch>
